File: java_distance_generator/trunk/src/java/main/com/topcoder/web/tc/distance/data/MemberDataAccessException.java
package com.topcoder.web.tc.distance.data;

import com.topcoder.util.errorhandling.BaseRuntimeException;
import com.topcoder.util.errorhandling.ExceptionData;

/**
 * <p>
 * This exception is thrown by <code>MemberDataAccess</code> implementations, when errors occur in the member retrieval
 * methods of the implementations.  The constructors of this exception wrap those of the base class.
 * </p>
 */
public class MemberDataAccessException extends BaseRuntimeException {

    /**
     * Creates a new, empty exception.
     */
    public MemberDataAccessException()
    {
        super();
    }

    /**
     * <p>
     * Creates a new exception containing the given message.
     * </p>
     *
     * @param message The message the exception will contain.
     */
    public MemberDataAccessException(String message) {
        super(message);
    }

    /**
     * <p>
     * Creates a new exception containing the given cause.
     * </p>
     *
     * @param cause The cause of the exception.
    */
    public MemberDataAccessException(Throwable cause) {
        super(cause);
    }

    /**
     * <p>
     * Creates a new exception containing the given message and cause.
     * </p>
     *
     * @param message The message the exception will contain.
     * @param cause The cause of the exception.
     */
    public MemberDataAccessException(String message, Throwable cause) {
        super(message, cause);
    }

    /**
     * <p>
     * Creates a new exception containing the given exception data.
     * </p>
     *
     * @param data The additional data to attach to the runtime exception.
     */
    public MemberDataAccessException(ExceptionData data) {
        super(data);
    }

    /**
     * <p>
     * Creates a new exception containing the given message and exception data.
     * </p>
     *
     * @param message The message the exception will contain.
     * @param data The additional data to attach to the runtime exception.
     */
    public MemberDataAccessException(String message, ExceptionData data) {
        super(message, data);
    }

    /**
     * <p>
     * Creates a new exception containing the given cause and exception data.
     * </p>
     *
     * @param cause The cause of the exception.
     * @param data The additional data to attach to the runtime exception.
     */
    public MemberDataAccessException(Throwable cause, ExceptionData data) {
        super(cause, data);
    }

    /**
     * <p>
     * Creates a new exception containing the given message, cause and exception data.
     * </p>
     *
     * @param message The message this exception will contain.
     * @param cause The cause of the exception.
     * @param data The additional data to attach to the runtime exception.
     */
    public MemberDataAccessException(String message, Throwable cause, ExceptionData data) {
        super(message, cause, data);
    }
}

File: dotnet_distance_generator/trunk/src/csharp/main/TopCoder/Web/Distance/Data/MemberDataAccessException.cs
/*
 * Copyright (c) 2008, TopCoder, Inc. All rights reserved.
 */

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace TopCoder.Web.Distance.Data
{
    /// <summary>
    /// This exception is thrown by IMemberDataAccess implementations, when errors
    /// occur in the member retrieval methods of the implementations. The constructors
    /// of this exception wrap those of the base class.
    /// </summary>
    [Serializable]
    public class MemberDataAccessException : ApplicationException
    {
        /// <summary>
        /// Creates a new, empty exception.
        /// </summary>
        public MemberDataAccessException() : base()
        {
        }

        /// <summary>
        /// Creates a new exception containing the given message.
        /// </summary>
        /// <param name="message">The message the exception will contain.</param>
        public MemberDataAccessException(string message) : base(message)
        {
        }

        /// <summary>
        /// Creates a new exception containing the given message and cause.
        /// </summary>
        /// <param name="message">The message the exception will contain</param>
        /// <param name="cause">The cause of the exception</param>
        public MemberDataAccessException(string message, Exception cause) : base(message, cause)
        {
        }

        /// <summary>
        /// Deserialization constructor.
        /// </summary>
        /// <param name="info">The serialization information used in deserialization</param>
        /// <param name="ctx">The streaming context used in deserialization</param>
        protected MemberDataAccessException(SerializationInfo info, StreamingContext ctx)
            : base(info, ctx)
        {
        }
    }
}
